src/components/dialog/useDialog.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState } from "react";
+
+declare global {
+  interface HTMLDialogElement {
+    showModal(): void;
+    close(): void;
+  }
+}
```

Comment:
Consider adding a reference/link to some documentation for this or replacing this with a third party package.

I wonder why we need to do this ourselves and how you came up with this.